cmake/FindHAL.cmake

```diff
@@ -1,251 +1,89 @@
+# Checking the parameters provided to the find_package(HAL ...) call
+# The expected parameters are families and or drivers
+################################################################################
+#Checking all the requested families before looking for drivers
+foreach(COMP ${HAL_FIND_COMPONENTS})
+    string(TOUPPER ${COMP} COMP_U)
+    string(REGEX MATCH "^STM32([FGHLMUW]P?[0-9BL])([0-9A-Z][0-9M][A-Z][0-9A-Z])?_?(M0PLUS|M4|M7)?.*$" COMP_U ${COMP_U})
+    if(CMAKE_MATCH_1) #Matches the family part of the provided STM32<FAMILY>[..] component
+        list(APPEND HAL_FIND_COMPONENTS_FAMILIES ${COMP})
+        message(TRACE "FindHAL: append COMP ${COMP} to HAL_FIND_COMPONENTS_FAMILIES")
+    else()
+        list(APPEND HAL_FIND_COMPONENTS_UNHANDLED ${COMP})
+    endif()
+endforeach()
 
-set(HAL_DRIVERS_WL
-    adc comp cortex crc cryp dac dma exti flash gpio gtzc hsem 
-    i2c i2s ipcc irda iwdg lptim pka pwr rcc 
-    rng rtc smartcard smbus spi subghz tim uart usart wwdg
-)
-set(HAL_EX_DRIVERS_WL
-    adc crc cryp dma flash i2c pwr rcc
-    rng rtc smartcard spi tim uart usart
-)
-set(HAL_LL_DRIVERS_WL
-    adc comp crc dac dma exti gpio i2c lptim lpuart pka pwr rcc 
-    rng rtc spi tim usart utils
-)
+# If no family requested look for all families
+if(NOT HAL_FIND_COMPONENTS_FAMILIES)
+    set(HAL_FIND_COMPONENTS_FAMILIES ${STM32_SUPPORTED_FAMILIES_LONG_NAME})
+endif()
 
-foreach(FAMILY_SUFFIX ${STM32_SUPPORTED_FAMILIES_SHORT_NAME})
-    list(APPEND HAL_DRIVERS ${HAL_DRIVERS_${FAMILY_SUFFIX}})
-    list(APPEND HAL_LL_DRIVERS ${HAL_LL_DRIVERS_${FAMILY_SUFFIX}})
+# Look for available drivers for all requested families
+foreach(family_comp ${HAL_FIND_COMPONENTS_FAMILIES})
+    string(TOUPPER ${family_comp} family_comp)
+    string(REGEX MATCH "^STM32([FGHLMUW]P?[0-9BL])([0-9A-Z][0-9M][A-Z][0-9A-Z])?_?(M0PLUS|M4|M7)?.*$" family_comp ${family_comp})
+    if(CMAKE_MATCH_1) #Matches the family part of the provided STM32<FAMILY>[..] component
+        set(FAMILY ${CMAKE_MATCH_1})
+    endif()
+    find_path(HAL_${FAMILY}_PATH
+        NAMES Inc/stm32${FAMILY_L}xx_hal.h
+        PATHS "${STM32_HAL_${FAMILY}_PATH}" "${STM32_CUBE_${FAMILY}_PATH}/Drivers/STM32${FAMILY}xx_HAL_Driver"
+        NO_DEFAULT_PATH
+        )
+    if(NOT HAL_${FAMILY}_PATH)
+        message(FATAL_ERROR "could not find HAL for family ${FAMILY}")
+    else()
+        set(HAL_${family_comp}_FOUND TRUE)
+    endif()
+    if(CMAKE_MATCH_1) #Matches the family part of the provided STM32<FAMILY>[..] component
+        get_list_hal_drivers(HAL_DRIVERS_${FAMILY} ${HAL_${FAMILY}_PATH} "hal")
+        get_list_hal_drivers(HAL_EX_DRIVERS_${FAMILY} ${HAL_${FAMILY}_PATH}  "ex")
+        get_list_hal_drivers(HAL_LL_DRIVERS_${FAMILY} ${HAL_${FAMILY}_PATH} "ll")
+        list(APPEND HAL_DRIVERS ${HAL_DRIVERS_${FAMILY}})
+        list(APPEND HAL_LL_DRIVERS ${HAL_LL_DRIVERS_${FAMILY}})
+    else()
+    endif()
 endforeach()
+message("P2H HAL drivers are ${HAL_DRIVERS}")
```

Comment:
This is not supposed to be left ^^

Reply:
I mean we could leave more trace for debug. Here we have access to different debug levels and by default it will not be printed. I you needed it during development then it must be usefull at least at a `TRACE` level
I explained same to SBE for data-logging.